lib/tapioca/dsl/compilers/url_helpers.rb

```diff
@@ -134,17 +143,21 @@ def gather_non_discoverable_includers
             end.freeze
           end
 
+          # Returns `true` if `mod` "directly" includes `helper`.
+          # For classes, this method will return false if the `helper` is included only by a superclass
           sig { params(mod: Module, helper: Module).returns(T::Boolean) }
           private def includes_helper?(mod, helper)
-            superclass_ancestors = []
+            ancestors = ancestors_of(mod)
 
-            if Class === mod
-              superclass = superclass_of(mod)
-              superclass_ancestors = ancestors_of(superclass) if superclass
+            unique_ancestors = if Class === mod && (superclass = superclass_of(mod))
+              # These ancestors are unique to `mod`, and exclude those in common with `superclass`.
+              ancestors.take(ancestors.count - ancestors_of(superclass).size)
+            else
+              ancestors
             end
 
-            ancestors = Set.new.compare_by_identity.merge(ancestors_of(mod)).subtract(superclass_ancestors)
-            ancestors.any? { |ancestor| helper == ancestor }
+            # Can't use a simple `include?(helper)` because of edge cases like `XPath`, whose `#==` operator misbehaves.
+            unique_ancestors.any? { helper.equal?(_1) }
```

Comment:
Honestly, I am not a fan of the numbered parameters for cases like this. Additionally, is there a special reason why we are using `equal?` instead of `==` as this used to be?

Reply:
Also, I'm kind of a huge fanboy for numbered params (well, at least `_1`, I'm less in favour if there are multiple), but I switched to to an explicitly named param instead :)